src/txmempool.cpp

```diff
@@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
     BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
         mapNextTx.erase(txin.prevout);
 
+    if (vTxHashes.size() > 1) {
+        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() - 1];
```

Comment:
Use `vTxHashes[it->vTxHashesIdx] = std::move(vTxHashes[vTxHashes.size() - 1])` or `std::swap(vTxHashes[it->vTxHashesIdx], vTxHashes[vTxHashes.size() - 1])` to avoid needing std::shared_ptr copy (which needs an atomic operation to increment the refcount).